### DISCUSSION TITLE
Adding a environment suffix to tags

### DISCUSSION THREAD

**Discussion by syphernl** [semantic-release/semantic-release#3059] [Q&A]
In the past we polyrepos for our apps and used [go-semrel-gitlab](https://juhani.gitlab.io/go-semrel-gitlab/) to handle the tagging/releases. 

In this setup the "app" repo contains `develop`/`staging`/`production` branches and the app-services were git-submodules based off `master`. 
Most projects have since moved to a monorepo structure (with the aforementioned branching approach) but a few of our projects cannot be changed to this due to their complexity.

In `go-semrel-gitlab` we had the option to create a [pre-release](https://juhani.gitlab.io/go-semrel-gitlab/pre-release/) such as
```
{{ (env "CI_ENVIRONMENT_SLUG") }},{{ seq }}
```
and it will produce versions/tags of form `v1.2.3-staging.1`.

This worked pretty well. Now, as `go-semrel-gitlab` hasn't been maintained for years I'd like to replace it with `semantic-release`. 
Normally we would configure the branches and set non-production branches to `prerelease: true` which end up creating a tag like that. 
However, in this particular instance we cannot do that since the services all have one branch.

Is there a way in `semantic-release` to handle this?

**Comment by travi** [comment]
you mention a number of different approaches here, but it is hard to understand what your target is. a couple of things that i can confirm to move the conversation maybe a small step further:
* semantic-release does not officially support monorepos. microrepos/polyrepos are the supported approach without the addition of unofficial/community supported addons
* the [pre-release workflow](https://semantic-release.gitbook.io/semantic-release/usage/workflow-configuration#pre-release-branches) requires the use of branches

**Reply by syphernl** [reply]
For our (newly structured) monorepo setup the semantic-release workflow works just fine as we only release the particular state of the app including the pre-release workflow based off our branches. 

Our poly repo approach works like this:
* The `app` repo contains git submodules that link to the particular service repos (e.g. `api`, `frontend` etc)
* The reference in the `app` repo is changed to point to the version one wants to deploy (e.g. `api`)
* A pipeline runs in the child project and as final step does a `release` stage (with `go-semrel-gitlab` currently)
  * Test release?  It gets tagged as `v1.2.3-test.X` (e.g. `v1.2.3-test.1`)
  * Staging release? It gets tagged as `v1.2.3-staging.X` (e.g. `v1.2.3-staging.1`)
  * Production release? Gets tagged as `v1.2.3`.
* Same thing happens in the app repo.

I was trying to implement the same approach with `semantic-release` but because of the legacy setup we cannot use a branch-based pre-release workflow and there doesn't seem to be a way to tell what the pre-release "tier" is (in our case the value of env var `CI_ENVIRONMENT_SLUG` which contains `test` / `staging` / `production`).
